### PR TITLE
fix: support for react-native 0.65

### PR DESCRIPTION
**Pull request by tido64**
### Description

Adds support for `react-native` 0.65.

### Platforms affected

- [x] Android
- [x] iOS
- [ ] macOS
- [ ] Windows

### Test plan

```sh
yarn set-react-version 0.65
cd example
rm ios/Podfile.lock  # not really necessary, just a precaution
yarn
pod install --project-directory=ios
yarn ios
```

| Home | App | App with Hermes |
|-|-|-|
| MicrosoftTeams-image (1) | MicrosoftTeams-image (2) | MicrosoftTeams-image (3) |

### CODE REVIEW

File: scripts/set-react-version.js [hunk 1]
```diff
@@ -38,6 +38,14 @@ const profiles = {
     "react-native-macos": undefined,
     "react-native-windows": "^0.64.11",
   },
+  0.65: {
+    "hermes-engine": "~0.8.1",
+    react: "17.0.2",
+    "react-native": "^0.65.0-0",
+    "react-native-codegen": "0.0.7",
```
Comment (tido64) [inline scripts/set-react-version.js:45]: For whenever 0.65 is released:
```suggestion
    "react-native": "^0.65.0",
```

File: example/react-native.config.js [hunk 2]
```diff
@@ -26,7 +26,30 @@ module.exports = {
       ),
     },
     ios: {
-      project: "ios/ReactTestApp-Dummy.xcodeproj",
+      project: (() => {
```
Comment (tido64) [inline example/react-native.config.js]: This needs to be ported to `scripts/configure.js` as well:
https://github.com/microsoft/react-native-test-app/blob/b72bb100d938d417fe7c0228693bcea29b84f37a/scripts/configure.js#L244-L246

File: scripts/set-react-version.js (hunk 1 above)
Comment (tido64) [reply scripts/set-react-version.js:45]: Seems like `react-native-codegen` will be necessary for 0.65. 😞

File: example/App.js [hunk 3]
```diff
@@ -52,21 +52,6 @@ const App = () => {
     backgroundColor: isDarkMode ? Colors.darker : Colors.lighter,
   };
 
-  const hermes = global.HermesInternal ? (
```
Comment (kelset) [inline example/App.js:55]: interesting, why did you remove the hermes section?

Comment (tido64) [reply example/App.js:55]: It was moved from the template to `NewAppScreen` in 0.64: https://github.com/facebook/react-native/commit/1c634a921887caf02f1b9fba6fd177c352f9ef78

**Review by kelset — APPROVED** [review]
